File: frontend/src/services/progressService.js
import { api } from '../boot/axios.js'

export default {

  async getCourseProgress(courseId) {
    return api.get(`progress/courses/${courseId}/progress`)
  },

  async completeLesson(lessonId) {
    return api.post(`progress/lessons/${lessonId}/complete`)
  },

  async getLessonsByCourse(courseId) {
    return api.get(`progress/courses/lesson/${courseId}`)
  }
}

File: backend/src/main/java/com/histack/ltplatform/service/ProgressService.java
package com.histack.ltplatform.service;

import java.util.List;

import com.histack.ltplatform.dto.ProgressResponseDto;
import com.histack.ltplatform.entity.LessonCompletions;
import com.histack.ltplatform.entity.Lessons;

public interface ProgressService {
	LessonCompletions completeLesson(Long lessonId, Long userId);
    ProgressResponseDto getCourseProgress(Long courseId, Long userId);
	List<Lessons> getLessonsByCourse(Long courseId);
}